gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -153,7 +161,10 @@ public void shutdownNow() {
 
   @Override
   public boolean awaitTermination(long duration, TimeUnit unit) throws InterruptedException {
-    return executor.awaitTermination(duration, unit);
+    if (shouldShutdownExecutor) {
+      return executor.awaitTermination(duration, unit);
+    }
+    return true;
```

Comment:
Should this wait for the future to complete with the specified time limit instead of returning true?